### ISSUE TITLE
sci-libs/flann not building with cmake 3.11 (workaround found)

### ISSUE THREAD

**Issue by awesomebytes** [ros/ros-overlay#591]
There is a bug filed in gentoo: https://bugs.gentoo.org/652594

That points to the issue in the flann repo: https://github.com/mariusmuja/flann/issues/369

And there is a patch to workaround it until upstream fixes it, can be added like:

```bash
# cd your EPREFIX
mkdir -p etc/portage/patches/sci-lib/flann-1.9.1
cd etc/portage/patches/sci-lib/flann-1.9.1
wget https://raw.githubusercontent.com/buildroot/buildroot/0c469478f64d0ddaf72c0622a1830d855306d51c/package/flann/0001-src-cpp-fix-cmake-3.11-build.patch
```

**Comment by pholthau** [comment]
I think the patch you mentioned is already in the main tree now:
https://github.com/gentoo/gentoo/blob/master/sci-libs/flann/flann-1.9.1-r1.ebuild

Actually, I needed some more patches to be applied, if compilation with cuda is needed, cf:
https://github.com/pholthau/gentoo-patches/tree/master/sci-libs/flann-1.9.1-r1

**Comment by pholthau** [comment]
@allenh1 yeah I know this is a gentoo base package. I just wanted to give a heads up to @awesomebytes and mabye others as I stumbled across this just yesterday and saw this issue. The repository I linked is actually symlinked to `/etc/portage/patches` during updates on one of my systems.
